packages/alfa-style/src/node/predicate/is-transparent.ts

```diff
@@ -23,33 +43,12 @@ export function isTransparent(
     return cache
       .get(device, Cache.empty)
       .get(context, Cache.empty)
-      .get(node, () => {
-        if (
-          isElement(node) &&
-          Style.from(node, device, context)
-            .computed("opacity")
-            .some((opacity) => opacity.value === 0)
-        ) {
-          return true;
-        }
-
-        for (const parent of node.parent(Node.flatTree)) {
-          if (
-            isText(node) &&
-            isElement(parent) &&
-            Style.from(parent, device, context)
-              .computed("color")
-              .some(
-                (color) => color.type === "color" && Color.isTransparent(color)
-              )
-          ) {
-            return true;
-          }
-
-          return isTransparent(parent);
-        }
-
-        return false;
-      });
+      .get(node, () =>
+        or(
+          and(isElement, isNotOpaque(device, context)),
+          and(isText, hasTransparentTextColor(device, context)),
+          (node: Node<string>) => node.parent(Node.flatTree).some(isTransparent)
```

Comment:
**Question:** why is the type annotation needed?
```suggestion
          (node) => node.parent(Node.flatTree).some(isTransparent)
```

Reply:
The type will be `unknown` otherwise, I'm actually not sure why

Reply:
Ah, yes, the predicate could be anything and `Cache` only expect a fully evaluated `boolean`…
Does it work with `test(or(…), node)`? Then the typing of `test` should force the predicate to be about `Node` 🤔 

(in any case, `Node` should be enough, `Node<string>` is the default.

Reply:
`test` didn't make a difference, but added it anyway to be consistent and used just `Node`